restful.go

```diff
@@ -199,7 +203,31 @@ func postRestfulQueryHelper(
 
 		var resultURL string
 		isSessionRenewed := false
+		noResult, _ := isAsyncMode(ctx)
```

Comment:
Should Restful as a wrapper on RESTFUL req/resp (transport layer, provide some hooks - 'Func'  at the most) be aware of that it has to write things to Rows or Result or even Rows/Result data structures that are specific to golang's sql module? Fill and return a wrapper object (e.g. QueryHandler) to include queryID, status, and statusChannel should be enough.  I think the original code that aimed to return queryId  only in async manner is simpler and more flexible.

Reply:
As discussed in the call today, the issue with returning a query handler object is that the user has no way to retrieve the `rows` or `result` object after. We can build a way for that to be possible, then our initial method of hanging on calls like `rows.Next()` is deemed moot, since that call depends on the async retrieval being completed. If we return a query handler and use the query ID to retrieve the result, the retrieval process will be kicked off only when the customer queries again. This method is more inline with existing code and APIs

Reply:
I thought I already answered the question on Rows.Next() and we agreed to hanging on Rows.Next() is the simple way to code.  What was the concern raised on blocking at Rows.Next()? I think we agree that to make ExcuteContext and QueryContext async is the goal, and this PR is to allow first query to return immediately, but if future callers want to get queryId for this query they have to wait for the queryId channel to return something.